lightly/api/download.py

```diff
@@ -25,27 +25,39 @@ def _check_av_available() -> None:
     if isinstance(av, Exception):
         raise av
 
-def download_image(url: str, session: requests.Session = None) -> PIL.Image.Image:
+def download_image(
+    url: str, 
+    session: requests.Session = None,
+    retry_fn: Callable = utils.retry,
+    request_kwargs: Dict = None,
+) -> PIL.Image.Image:
     """Downloads an image from a url.
 
     Args:
         url: 
             The url where the image is downloaded from.
         session: 
             Session object to persist certain parameters across requests.
+        retry_fn:
+            Retry function that handles failed downloads.
+        request_kwargs:
+            Additional parameters passed to requests.get().
 
     Returns:
         The downloaded image.
 
     """
-    def load_image(url, req):
-        with req.get(url=url, stream=True) as response:
+    request_kwargs = request_kwargs or dict(stream=True, timeout=10)
```

Comment:
Here I would rather use `request_kwargs.set_default()`. Otherwise, there will be no `stream=True` if` {"timeout": 2}` is passed.